libraries/standard/mqtt/doc/doxygen/pages.txt

```diff
@@ -0,0 +1,257 @@
+/**
+@mainpage Overview
+@brief MQTT 3.1.1 client library.
+
+> MQTT stands for MQ Telemetry Transport. It is a publish/subscribe, extremely simple and lightweight messaging protocol, designed for constrained devices and low-bandwidth, high-latency or unreliable networks. The design principles are to minimise network bandwidth and device resource requirements whilst also attempting to ensure reliability and some degree of assurance of delivery. These principles also turn out to make the protocol ideal of the emerging "machine-to-machine" (M2M) or "Internet of Things" world of connected devices, and for mobile applications where bandwidth and battery power are at a premium.
+<span style="float:right;margin-right:4em"> &mdash; <i>Official description of MQTT from [mqtt.org](http://mqtt.org)</i></span><br>
+
+This MQTT library implements the client side of the MQTT 3.1.1 protocol. This library is optimized for resource constrained embedded devices. Features of this library include:
+- Fully synchronous API, to allow applications to completely manage their concurrency and multi-threading method.
+- Operations on fixed buffers, so that applications may control their memory allocation strategy.
+- Scalable performance and footprint. The [configuration settings](@ref mqtt_config) allow this library to be tailored to a system's resources.
+
+Please see https://github.com/aws/aws-iot-device-sdk-embedded-C/ for example code demonstrating integration with TLS.
```

Comment:
Should this link to the specific directory of the tls demo?

Reply:
that would be better

Reply:
We will also have to make sure to update this link after everything moves to master